include/uxr/agent/transport/endpoint/CustomEndPoint.hpp

```diff
@@ -450,7 +450,7 @@ class CustomEndPoint
     }
 
 private:
-    std::map<const std::string, Member> members_;
```

Comment:
I have understood the other changes, but why this?

Reply:
> I have understood the other changes, but why this?

@pablogs9 Hi Pablo.
I think this depends a lot on how ```std::map``` is implemented in ```libc++``` of ```llvm```. Take one of the errors as an example:
```
/home/joel/android-ndk-r23/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/c++/v1/map:708:17: error: no viable overloaded '='
        __ref() = __v.__get_value();
        ~~~~~~~ ^ ~~~~~~~~~~~~~~~~~
```
and take a look at the ```map``` implementation:
```cpp
template <class _Key, class _Tp>
struct __value_type
{
    typedef _Key                                     key_type;
    typedef _Tp                                      mapped_type;
    typedef pair<const key_type, mapped_type>        value_type;
    typedef pair<key_type&, mapped_type&>            __nc_ref_pair_type;
    typedef pair<key_type&&, mapped_type&&>          __nc_rref_pair_type;

private:
    value_type __cc;

public:
    _LIBCPP_INLINE_VISIBILITY
    value_type& __get_value()
    {
#if _LIBCPP_STD_VER > 14
        return *_VSTD::launder(_VSTD::addressof(__cc));
#else
        return __cc;
#endif
    }

    _LIBCPP_INLINE_VISIBILITY
    const value_type& __get_value() const
    {
#if _LIBCPP_STD_VER > 14
        return *_VSTD::launder(_VSTD::addressof(__cc));
#else
        return __cc;
#endif
    }

    _LIBCPP_INLINE_VISIBILITY
    __nc_ref_pair_type __ref()
    {
        value_type& __v = __get_value();
        return __nc_ref_pair_type(const_cast<key_type&>(__v.first), __v.second);
    }

    _LIBCPP_INLINE_VISIBILITY
    __nc_rref_pair_type __move()
    {
        value_type& __v = __get_value();
        return __nc_rref_pair_type(
            _VSTD::move(const_cast<key_type&>(__v.first)),
            _VSTD::move(__v.second));
    }

    _LIBCPP_INLINE_VISIBILITY
    __value_type& operator=(const __value_type& __v)
    {
        __ref() = __v.__get_value();  // Here is where the build error is located.
        return *this;
    }
   ......
}
```
To make a assign constructor, ```__ref()``` generate a ```pair``` which intentionally removes the ```const``` added by ```value_type```. But if the ```key_type``` itself is still a ```const```, this assignment fails because we simply don't reassign to a ```const```.

Even with implementations other than ```libc++```, I don't think callers of ```std::map``` should ```const```ify the key string to make it immutable. ```std::map``` itself does the job where necessary, such as the ```std::map::value_type```.

Here are two related discussions:
https://stackoverflow.com/questions/17638154/what-is-difference-between-const-and-non-const-key
https://lists.llvm.org/pipermail/cfe-dev/2011-July/015926.html

Reply:
@Acuadros95 you ok with this?

Reply:
Yes, this shouldn't be a problem.
In any case, let's wait for the integration tests output: https://github.com/eProsima/Micro-XRCE-DDS/pull/111